### PR TITLE
Fix record intersection configurable error

### LINKED ISSUE #40277: [Bug]: Program won't start with default value set for array/map/table field in configurable record
### Description

When a default value exists for a field of array, map, or table type in a configurable record, the Ballerina program fails to start with the following error.

```
error: {ballerina/lang.map}InherentTypeViolation {"message":"invalid value for record field 'url': expected value of type '(string[] & readonly)', found 'string[]'"}
```

### Steps to Reproduce

Here is the code snippet. 

```ballerina
type Endpoint record {|
    string basePath;
    string[] url = [];
|};

configurable Endpoint[] endpoints = [];
```

Config.toml
```toml
[[endpoints]]
basePath = "/api/entitlement-service"
url = ["https://abc.com/api/entitlement-service"]

```

### Affected Version(s)

2201.5.0

### OS, DB, other environment details and versions

_No response_

### Related area

-> Runtime

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by HindujaB**
## Purpose
$subject 

Fixes #40277 

## Approach
Fixing the default value check by creating a record value with mutable type and cloning that value to a `readonly` value.

## Samples
```ballerina
type Endpoint record {|
    string basePath;
    string[] url = [];
|};

configurable Endpoint[] endpoints = [];
```
Config.toml
```toml
[[endpoints]]
basePath = "/api/entitlement-service"
url = ["https://abc.com/api/entitlement-service"]
```

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [x] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/toml/ConfigValueCreator.java [hunk 1]
```diff
@@ -254,29 +256,32 @@ private Object getElementValue(Type elementType, TomlNode tomlValueNode) {
...
 
     private BMap<BString, Object> createRecordValue(TomlNode tomlNode, Type type) {
-        RecordType recordType;
-        String recordName;
-        if (type.getTag() == TypeTags.RECORD_TYPE_TAG) {
-            recordName = type.getName();
-            recordType = (RecordType) type;
+        RecordType mutableType;
+        Optional<IntersectionType> intersectionType = ((IntersectableReferenceType) type).getIntersectionType();
+        if (intersectionType.isPresent()) {
```
Comment (warunalakshitha) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/toml/ConfigValueCreator.java:261]: Shall we add a comment here. And create and issue to revert this once we fix the record init

Comment (HindujaB) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/toml/ConfigValueCreator.java:261]: As per the spec issue conclusion (https://github.com/ballerina-platform/ballerina-spec/issues/544), this is the suggested way of creating the intersection values. 
I'll add a comment including the value creation-related info.

**Review by gabilang — APPROVED** [review]
LGTM

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Pull request merged**